### ISSUE TITLE
Search regression: my cats / my dogs

### ISSUE THREAD

**Issue by fbanados** [UAlbertaALTLab/crk-db#122]
there is some issue (likely associated with the English Phrase FST not adding an `+A` tag) that prevents the dev version from correctly providing an inflected form when searching my cats/my dogs.  However, the FST behaviours are equivalent, so a different justification for the failure must be identified to make the problem reproducible.  Needs fixing.

**Comment by fbanados** [comment]
e.g. entry for cats should have:

`{
  "analysis": [
    [],
    "minôs",
    [
      "+N",
      "+A",
      "+Sg"
    ]
  ],
...`

**Comment by fbanados** [comment]
There will be several notes to add about these examples, but for starters,
we should add analyses to all entries with a `+A` tag.

**Comment by fbanados** [comment]
After matching against the referenced https://github.com/giellalt/lang-crk/blob/main/tools/shellscripts/add-explicit-fields-to-crkeng.sh, issues with `+A` suffix are solved.  However, there is still work remaining:

- [ ] Classify, categorize and decide on extra analysis:  It is likely that these should sometimes not be added if they weren't added before.
- [ ] Classify, categorize, and provide a fix to add an analysis on the ~1k pending entries that still do not include an analysis (and they did before)

**Comment by fbanados** [comment]
To fix regression, after ensuring that analysis includes `+A`, **a restart of the docker container is required**, otherwise search results are not properly sorted (that is, `cosine_vector_distance` is `null` instead of `0.0` in some cases, leading to incorrect results).  That is a separate bug.

**Comment by fbanados** [comment]
Most missing entries were Ipc, and a buggy comparison where `IPC != Ipc`.  The 30 leftover are issues with the FST, that requires linguistic feedback. That is mostly heads that are no longer recognized by the strict FST (`'nipâskâkow', 'mac-âyiwiwin', 'mac-âtocikêw', 'oski-ôsi', 'mac-âcimoskiw', 'mac-âcimowin', 'mac-âcimow', 'osk-âya', 'osk-âyi', 'mac-âyiwiw', 'mihtos', 'osk-âyisis', 'mac-âcimiwêw', 'mistiko-mahkahkos', 'waskway-ôsi', 'mistahi-ôsi', 'mac-âyisiwiw', 'kôhkomipaninaw', 'nipêskâkow', 'môhkocikêwikamik', 'pîhtawêwayiwinisa', 'wâsitêpimâkanihkêw', 'mac-âcimêw', 'âpihtawakimâw',  'mêstakimâw', 'mac-âyisiw', 'iskotêw-ôsi', 'osk-âyis'` or heads where the strict analysis produces multiple equal analyses (`'akik', 'okosisimâw'`). The entry in CW for `kôhkomipaninaw` has a different POS than that produced by the analysis of the FST.  Seems that none of these entry differences have an impact on the presentation in the dictionary, but should be double-checked by a linguist.

**Comment by aarppe** [comment]
As for many of the other elements such as _mac-âyiwiwin_, they are a case where there is orthographical variation at the preverb/prenoun-stem junction, based on reduction in speech. The full form would be _maci-âyiwiwin_, but because the stem starts with a vowel the preverb-final `-i-` is often dropped.

We started a discussion with Arok about how to deal with these forms. One would be inclined to choose one variant as the more standard form, and then accept the variants (rather than creating two FST lemmas, if one enumerates both in the LEXC file for stems.) Currently these are sort-of catched by the script, in that the `\fststem` field is marked, cf.
```

\sro mac-âyiwiwin
...
\ps NI-1
\def being bad, being mean, being wicked; doing evil; having a bad temper; being a dangerous being
\stm maci-ayiwiwin-
\fststem CHECK:maci-ayiwiwinw?- OLD:mac-âyiwiwin

```

**Comment by fbanados** [comment]
The FSTs seem to be detecting the variation, but from your comment it sounds like it may need to be the opposite of this:
![Screenshot 2024-07-24 at 5 14 51 PM](https://github.com/user-attachments/assets/49bd886f-532b-4b1f-9162-81388bcb2bf9)

Also, several of the other entries you mentioned as inflected forms (inanimate actor) of TA verbs are already detected as forms of.  The ones that are not link to an empty paradigm.

**Comment by aarppe** [comment]
Yes, I'd need to revise some parts of how the FST is generated for these sandhi forms.
